src/object/schema_manager.c

```diff
@@ -5767,6 +5767,74 @@ sm_att_info (MOP classop, const char *name, int *idp, TP_DOMAIN ** domainp, int
   return error;
 }
 
+
+/*
+ * sm_find_key_index()
+ *   return: Pointer to B-tree ID variable
+ *           has "att_name" as first key attribute.
+ *   classop(in): class object
+ *   att_names(in):
+ *   btid(out):
+ */
+
+BTID *
+sm_find_key_index (MOP classop, char *att_name, BTID * btid)
```

Comment:
Because of the num_atts of sm_find_index() , it seems that there is a parameter to check only a specific column. If it doesn't work as intended, wouldn't it be better to fix a bug in the existing sm_find_index() rather than adding sm_find_key_index()?

Reply:
The purpose of sm_find_index() is to find an index includes specific attribute which is given by function argument.
The routine finds an index having attribute as first and also second column. This is not a bug. It's just a purpose of the function. We can fix or refactor the function to add a new purpose to get an index having attribute as first column only for multi-key index but This function is called from many routine. I think it would be better to make a new function.

Reply:
There is posted a test case which is written by cci-driver on CBRD-24436 as like count(*). The code hase 3 tranaction group having 50 thread running , which are DML (insert/delete), MIN (select min), MAX (select MAX).

Reply:
Ok, I fixed it as you commented.